android/app/src/main/java/com/mattermost/rnbeta/CustomPushNotification.java

```diff
@@ -222,69 +229,46 @@ protected void postNotification(int id, Notification notification) {
             largeIconResId = res.getIdentifier("ic_launcher", "mipmap", packageName);
         }
 
-        if (numberString != null) {
-            CustomPushNotification.badgeCount = Integer.parseInt(numberString);
+        if (badge != null) {
+            CustomPushNotification.badgeCount = Integer.parseInt(badge);
             ApplicationBadgeHelper.instance.setApplicationIconBadgeNumber(mContext.getApplicationContext(), CustomPushNotification.badgeCount);
+        } else {
+            // HERE ADD THE DOT INDICATOR STUFF
         }
 
-        int numMessages = getMessageCountInChannel(channelId);
+        Notification.MessagingStyle messagingStyle = new Notification.MessagingStyle("You");
+        if (title != null && !title.startsWith("@")) {
+            messagingStyle
+                    .setConversationTitle(title);
+        }
+
+        List<Bundle> bundleArray = channelIdToNotification.get(channelId);
+        List<Bundle> list;
+        if (bundleArray != null) {
+            list = new ArrayList<Bundle>(bundleArray);
+        } else {
+            list = new ArrayList<Bundle>();
+        }
+
+        for (Bundle data : list) {
+            String message = data.getString("message");
+            if ((title != null && !title.startsWith("@") || title == null)) {
```

Comment:
This could be shortened slightly
```suggestion
            if (title == null || !title.startsWith("@")) {
```